### PR TITLE
fix(unpin): bug caused us to not be able to unpin to 0

### PR DESCRIPTION
**Pull request by dreynaud**
0 was being used in the Elvis operator but it is not Groovy-truthy.
Fix that to an actual null check.

### CODE REVIEW

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategySupportSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import spock.lang.Subject
+import spock.lang.Unroll
+
+class ResizeStrategySupportSpec extends Specification {
+  @Subject
+  ResizeStrategySupport resizeStrategySupport
+
+  @Unroll
+  def "test min logic in performScalingAndPinning() with unpinMin=#unpinMin originalMin=#originalMin savedMin=#savedMin"() {
+    resizeStrategySupport = new ResizeStrategySupport()
```
Comment (ajordens) [inline orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategySupportSpec.groovy]: should be in a `given:` block

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategySupportSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    [min: 1, max: 3, desired: 2] | false    | 1           | null     || [min: 1, max: 3, desired: 2]
+    [min: 1, max: 3, desired: 2] | true     | 1           | null     || [min: 1, max: 3, desired: 2]
+    [min: 1, max: 3, desired: 2] | true     | 2           | null     || [min: 1, max: 3, desired: 2] // won't unpin to a higher min 2
+    [min: 1, max: 3, desired: 2] | true     | 0           | null     || [min: 0, max: 3, desired: 2]
+    [min: 1, max: 3, desired: 2] | true     | null        | 2        || [min: 1, max: 3, desired: 2]
+    [min: 1, max: 3, desired: 2] | true     | 0           | 2        || [min: 0, max: 3, desired: 2]
+    [min: 1, max: 3, desired: 2] | true     | null        | 0        || [min: 0, max: 3, desired: 2]
+  }
+
+  private Map toMap(ResizeStrategy.Capacity capacity) {
```
Comment (ajordens) [inline orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategySupportSpec.groovy]: don't see these methods called anywhere

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategySupportSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    stage.context = [
+      unpinMinimumCapacity: unpinMin,
+      source: [
+          serverGroupName: "app-v000"
+      ],
+      "originalCapacity.app-v000": [
+          min: originalMin
+      ],
+      savedCapacity: [
+          min: savedMin
```
Comment (ajordens) [inline orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategySupportSpec.groovy]: I don't think the test is catching the `savedCapacity` path?

Comment (dreynaud) [reply orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategySupportSpec.groovy]: the last test case does:

        [min: 1, max: 3, desired: 2] | true     | null        | 0        || [min: 0, max: 3, desired: 2] // picks the savedMin value

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategySupportSpec.groovy (hunk 2 above)
Comment (dreynaud) [reply orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategySupportSpec.groovy]: good catch, I forgot that I rewrote the test so these were not needed anymore

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategySupportSpec.groovy (hunk 3 above)
Comment (ajordens) [reply orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategySupportSpec.groovy]: Doh, sorry missed that!

**Pull request merged**